ignition/what-is-ignition.md

```diff
@@ -66,7 +60,9 @@ experince crippled because the metadata for their platform isn't supported. It
 is possible to write a [custom metadata agent][custom agent] to fetch the
 necessary data.
 
-[custom agent]: examples.md#custom-metadata-agent
+### How Ignition is Being Executed ##
+
+GRUB checks the disks for the dummy disk UUID and sets `coreos.first_boot=1` parameter accordingly. This parameter is being processed by [systemd-generators] every boot inside [initramfs] and whenever this parementer is set systemd adds `ignition.target` as one of the `initrd.target` requirement to run Ignition. In case when you use [PXE][supported platforms] deployment `coreos.first_boot=1` parameter needs to be set manually otherwise Ignition won't run.
```

Comment:
On boot, GRUB checks the disks for the dummy disk UUID (`00000000-0000-0000-0000-000000000000`) and sets `coreos.first_boot=1` if it is found. This parameter is then processed by a [systemd-generator] in the [initramfs] and if the parameter is set to non-zero, `ignition.target` is set as a dependency of `initrd.target`, causing Ignition to run.

Note that [PXE][supported platforms] deployments don't using GRUB to boot, so the `coreos.first_boot=1` parameter will need to be added to the boot arguments in order for Ignition to run.